### DISCUSSION TITLE
Module API 500 Debug

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#733] [General] [answered]
when I query my API for data, I get 500 errors that render like HTML.  I don't think my module is loading correctly.  I'm having a tough time debugging this and there is nothing in the logs.  Has anyone ran into this and/or could suggest some debugging.  I've tried setting breakpoint in the controller, but it never gets hit.  I think it's a service level problem.

![image](https://user-images.githubusercontent.com/311063/92509801-b4a54900-f1bf-11ea-82d7-670a5c66e18b.png)

**Comment by hishamco** [comment]
Which API that you are trying to request?

**Reply by markdav-is** [reply]
my own custom api that was part of the module

**Reply by hishamco** [reply]
I thought it's Oqtane one, please show us how your API is look like or provide a repo to reproduce

**Reply by markdav-is** [reply]
I'll work up a sample!  it might be tricky without a database, but.... should give a different error.

**Comment by mikecasas** [comment]
@nohorse  Try removing all auth and see if your break point gets hit.

**Comment by nicpitsch** [comment]
> ...I've tried setting breakpoint in the controller, but it never gets hit. I think it's a service level problem.

The error might - as you suppose - happen earlier, with the service already. I would suggest to set a breakpoint for the response in *GetJsonAsync* method in *ServiceBase* class in Oqtane.Client (within Oqtane framework solution, not your module). Start Oqtane Debugging with F5.

**Comment by markdav-is** [comment, marked as answer]
Thanks for you for assistance.  I ended up rebuilding the module from scratch and it works in debug.
